src/datastore/pg-write-store.ts

```diff
@@ -256,20 +255,30 @@ export class PgWriteStore extends PgStore {
       }
       if ((await this.updateBlock(sql, data.block)) !== 0) {
         await this.updateMinerRewards(sql, data.minerRewards);
-        for (const entry of batchedTxData) {
-          await this.updateTx(sql, entry.tx);
-          await this.updateStxEvents(sql, entry.tx, entry.stxEvents);
-          await this.updatePrincipalStxTxs(sql, entry.tx, entry.stxEvents);
-          await this.updateSmartContractEvents(sql, entry.tx, entry.contractLogEvents);
-          await this.updatePoxSyntheticEvents(sql, entry.tx, 'pox2_events', entry.pox2Events);
-          await this.updatePoxSyntheticEvents(sql, entry.tx, 'pox3_events', entry.pox3Events);
-          await this.updatePoxSyntheticEvents(sql, entry.tx, 'pox4_events', entry.pox4Events);
-          await this.updateStxLockEvents(sql, entry.tx, entry.stxLockEvents);
-          await this.updateFtEvents(sql, entry.tx, entry.ftEvents);
-          await this.updateNftEvents(sql, entry.tx, entry.nftEvents);
-          await this.updateSmartContracts(sql, entry.tx, entry.smartContracts);
-          await this.updateNamespaces(sql, entry.tx, entry.namespaces);
-          await this.updateNames(sql, entry.tx, entry.names);
+        if (batchedTxData.length > 0) {
+          const q = new PQueue({ concurrency: 4 });
```

Comment:
Can you make `concurrency` configurable via ENV? Or why is it set to 4? Curious if you did any benchmarks or how did you get to that number

Reply:
Also, can you initialize the `PQueue` outside this block so we can reuse it across blocks? Will save CPU cycles especially during event replays

Reply:
4 is better than 1, and a default value not being too radical if someone is running postgres instance on non-SSD disk.

Added the env entry, please help me review the wording.

Reply:
> initialize the PQueue outside this block

It's not safe, `await q.onIdle()` will wait for all tasks in the queue, which might come from another process.

Besides, the queue initialization overhead is small compares to all the tasks during event replays.

Reply:
We process blocks in series so they would not overlap across blocks, but I think we can start with this approach as we test this PR in an env.